### PR TITLE
Test HTTP/2 `KeepAliveManager` with `DuplexChannel`

### PR DESCRIPTION
**Pull request by idelpivnitskiy**
Motivation:

`KeepAliveManagerTest` covers only non-duplex `Channel` use-case. However, `KeepAliveManager` state machine is a lot more complex for `DuplexChannel`. We need to enhance test coverage for the most used case.

Modifications:
- Parametrize `KeepAliveManagerTest` to test with duplex and non-duplex `Channel`, add more assertions;
- Modify `H2ParentConnectionContext` and `KeepAliveManager` to accept `Http2StreamChannel` instead of `Channel`;
- `KeepAliveTest`: fix `clientBuilder.initializeHttp` override issue (`CLIENT_CTX` is not set), do some other minor refactoring;

Result:

Enhanced test coverage for `KeepAliveManager`.

### CODE REVIEW

File: servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java [hunk 1]
```diff
@@ -47,41 +59,54 @@
 
 class KeepAliveManagerTest {
 
+    private final BlockingQueue<ScheduledTask> scheduledTasks = new LinkedBlockingQueue<>();
```
Comment (daschl) [inline servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java:62]: I know it wasn't done in the original code / this refactoring, but would it make sense in the BeforeEach method (or setUp) to clear this queue to have every test start with a deterministic fresh state?

File: servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java [hunk 2]
```diff
@@ -338,4 +396,33 @@ private static final class ScheduledTask {
...
+    private static final class KeepAliveManagerHandler extends ChannelInboundHandlerAdapter {
+        private KeepAliveManager keepAliveManager;
+
+        void keepAliveManager(KeepAliveManager keepAliveManager) {
+            this.keepAliveManager = requireNonNull(keepAliveManager);
+        }
+
+        @Override
+        public void channelInactive(final ChannelHandlerContext ctx) {
+            keepAliveManager.channelClosed();
```
Comment (daschl) [inline servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java:409]: IIRC handlerRemoved and channelInactive can be both called one after another in the pipeline, is it necessary to add an "already closed" check in `KeepAliveManager#channelClosed`? The implementation of the method looks fine to me side-effect wise, just double checking with you if it's ok.

File: servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java [hunk 3]
```diff
@@ -338,4 +396,33 @@ private static final class ScheduledTask {
...
+        }
+
+        @Override
+        public void userEventTriggered(final ChannelHandlerContext ctx, final Object evt) throws Exception {
+            if (evt == ChannelInputShutdownReadComplete.INSTANCE || evt == SslCloseCompletionEvent.SUCCESS) {
+                keepAliveManager.channelInputShutdown();
+            } else if (evt == ChannelOutputShutdownEvent.INSTANCE) {
+                keepAliveManager.channelOutputShutdown();
+            } else {
+                release(evt);
```
Comment (daschl) [inline servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java:424]: Should any other user-triggered events cause failure? (i.e. they are unexpected?)

File: servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java (hunk 2 above)
Comment (idelpivnitskiy) [reply servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java:409]: Thanks for calling it out. It looks idempotent to me. It only overrides the states and cancels if there are any futures.

File: servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java (hunk 3 above)
Comment (idelpivnitskiy) [reply servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java:424]: I don't think so. My understanding is that user events are fired as best effort to notify a pipeline about something. Other handlers are free to ignore those events if they do not depend on them. 

This handler is a simplification of what we have in `H2ParentConnectionContext.AbstractH2ParentConnection`.

**Review by daschl — APPROVED** [review]

**Pull request merged**